ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/MLSdkAsyncHttpResponseHandler.java

```diff
@@ -80,6 +81,23 @@ public void onHeaders(SdkHttpResponse response) {
         SdkHttpFullResponse sdkResponse = (SdkHttpFullResponse) response;
         log.debug("received response headers: " + sdkResponse.headers());
         this.statusCode = sdkResponse.statusCode();
+        if (MapUtils.isEmpty(sdkResponse.headers())) {
+            return;
+        }
+        List<String> errorsInHeader = sdkResponse.headers().get("x-amzn-ErrorType");
+        if (errorsInHeader == null || errorsInHeader.isEmpty()) {
+            return;
+        }
+        boolean containsThrottlingException = errorsInHeader.stream().anyMatch(str -> str.startsWith("ThrottlingException"));
+        if (containsThrottlingException) {
+            actionListener
+                .onFailure(
+                    new OpenSearchStatusException(
```

Comment:
How about create a exception class like RemoteThrottlingException, so that we can identify the exception type quickly without checking its string content. For example, we need to check its type to implement retry logic https://github.com/opensearch-project/ml-commons/issues/2438

Reply:
Fair enough. But if I miss declaring this new class, please add it in your next PR for the retry implementation.